File: app/src/main/java/com/worldventures/dreamtrips/wallet/service/command/profile/UpdateProfileModule.java
package com.worldventures.dreamtrips.wallet.service.command.profile;

import com.worldventures.dreamtrips.wallet.service.SmartCardInteractor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.techery.janet.Janet;

@Singleton
@Module(
      injects = {
            UpdateSmartCardUserCommand.class,
            RevertSmartCardUserUpdatingCommand.class,
            RetryHttpUploadUpdatingCommand.class
      },
      library = true, complete = false)
public class UpdateProfileModule {

   @Provides
   @Singleton
   UpdateDataHolder updateDataHolder() {
      return new UpdateDataHolder();
   }

   @Provides
   @Singleton
   UpdateProfileManager updateProfileManager(
         Janet janetApi,
         SmartCardInteractor smartCardInteractor,
         UpdateDataHolder updateDataHolder) {
      return new UpdateProfileManager(janetApi, smartCardInteractor, updateDataHolder);
   }
}

File: wallet/src/main/java/com/worldventures/wallet/service/profile/UpdateProfileModule.kt
package com.worldventures.wallet.service.profile

import com.worldventures.wallet.di.WalletJanetModule
import com.worldventures.wallet.service.SmartCardInteractor
import dagger.Module
import dagger.Provides
import io.techery.janet.Janet
import io.techery.mappery.MapperyContext
import javax.inject.Named
import javax.inject.Singleton

@Singleton
@Module(injects = arrayOf(
      UpdateSmartCardUserCommand::class,
      RevertSmartCardUserUpdatingCommand::class,
      RetryHttpUploadUpdatingCommand::class,
      UpdateSmartCardUserPhotoCommand::class),
      library = true, complete = false)
class UpdateProfileModule {

   @Provides
   @Singleton
   fun updateProfileManager(@Named(WalletJanetModule.JANET_WALLET) janetApi: Janet,
                            mapperyContext: MapperyContext,
                            smartCardInteractor: SmartCardInteractor): UpdateProfileManager =
         UpdateProfileManager(janetApi, mapperyContext, smartCardInteractor)
}
